File: packages/typescript/src/resources/memories.ts
import { BaseResource } from './base.js';
import type {
  Memory,
  MemoryCreateInput,
  MemorySearchInput,
  MemoryUpdateInput,
  QueryParams,
} from '../types.js';

export class MemoriesResource extends BaseResource {
  list(query?: QueryParams): Promise<Memory[]> {
    return this.client.request<Memory[]>(
      query
        ? { path: 'memories', query }
        : { path: 'memories' },
    );
  }

  get(id: string): Promise<Memory> {
    return this.client.request<Memory>({ path: `memories/${encodeURIComponent(id)}` });
  }

  create(body: MemoryCreateInput): Promise<Memory> {
    return this.client.request<Memory, MemoryCreateInput>({
      path: 'memories',
      method: 'POST',
      body,
    });
  }

  update(id: string, body: MemoryUpdateInput): Promise<Memory> {
    return this.client.request<Memory, MemoryUpdateInput>({
      path: `memories/${encodeURIComponent(id)}`,
      method: 'PUT',
      body,
    });
  }

  delete(id: string): Promise<void> {
    return this.client.request<void>({
      path: `memories/${encodeURIComponent(id)}`,
      method: 'DELETE',
      responseType: 'void',
    });
  }

  search(body: MemorySearchInput): Promise<Memory[]> {
    return this.client.request<Memory[], MemorySearchInput>({
      path: 'memories/search',
      method: 'POST',
      body,
    });
  }

  getStm(agentId: string): Promise<Memory[]> {
    return this.client.request<Memory[]>({
      path: `memories/stm/${encodeURIComponent(agentId)}`,
    });
  }

  get_stm(agentId: string): Promise<Memory[]> {
    return this.getStm(agentId);
  }
}

File: packages/flutter/lib/src/resources/memories.dart
/// Memories resource — `/api/memories`.
library;

import '../models.dart';
import 'base.dart';

class MemoriesResource extends BaseResource {
  const MemoriesResource(super.client);

  Future<List<Memory>> list({
    String? agentId,
    String? scope,
  }) async {
    final query = <String, String>{};
    if (agentId != null) query['agent_id'] = agentId;
    if (scope != null) query['scope'] = scope;
    final data = await client.request(
      'GET',
      'memories',
      queryParameters: query.isEmpty ? null : query,
    );
    if (data is List) {
      return data.whereType<Map<String, dynamic>>().map(Memory.fromJson).toList();
    }
    return const [];
  }

  Future<Memory> get(String id) async {
    final data = await client.request('GET', 'memories/$id');
    return Memory.fromJson(data as Map<String, dynamic>);
  }

  Future<Memory> create(MemoryCreate payload) async {
    final data = await client.request('POST', 'memories', body: payload.toJson());
    return Memory.fromJson(data as Map<String, dynamic>);
  }

  Future<Memory> update(String id, MemoryUpdate payload) async {
    final data = await client.request('PUT', 'memories/$id', body: payload.toJson());
    return Memory.fromJson(data as Map<String, dynamic>);
  }

  Future<void> delete(String id) => client.request('DELETE', 'memories/$id');

  Future<List<Memory>> search(MemorySearchRequest payload) async {
    final data = await client.request('POST', 'memories/search', body: payload.toJson());
    if (data is List) {
      return data.whereType<Map<String, dynamic>>().map(Memory.fromJson).toList();
    }
    return const [];
  }

  Future<List<Memory>> getStm(String agentId) async {
    final data = await client.request('GET', 'memories/stm/$agentId');
    if (data is List) {
      return data.whereType<Map<String, dynamic>>().map(Memory.fromJson).toList();
    }
    return const [];
  }
}
